### DISCUSSION TITLE
High Noise using Partially Hidden Emitters?

### DISCUSSION THREAD

**Discussion by coult099** [mitsuba-renderer/mitsuba3#554] [Q&A] [answered]
Having some issues rendering scenes where an area emitter is partially hidden with the `path` integrator, as shown here: 
![veach-bidir_test_cuda_ad_spectral_spp1024_2023-02-22](https://user-images.githubusercontent.com/115170151/220798757-21010560-0785-4334-8b7a-e29180873732.png)

I get this noise using the `prb` integrator as well, but not with `direct`. Additionally, I get less noise as I move the area emitters out of the lamp meshes, with none of this noise when the area emitters are exposed completely.

Any insight would be appreciated! The `path` integrator is suggested by the [creator of the scene](https://benedikt-bitterli.me/resources/).

**Comment by njroussel** [comment, marked as answer]
The `direct` integrator should produce quite a different output, no?

This has to do with the depth of the ray paths and next event estimation. Take a look at the very top of the [integrator plugin documentation page](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_integrators.html#id11) and [this chapter](https://pbr-book.org/3ed-2018/Light_Transport_I_Surface_Reflection/Path_Tracing#) of PBRT.
Try to reason about how the rays are constructed: they start from the sensors, hit some surface and scatter into a new direction. At every bounce/hit, we try to connect the ray directly to an emitter. It should be obvious that, for this scene specifically, we can only rarely connect it to an emitter. In fact, you can see this visually in the output of the `direct` integrator.
However, if you move the emitter such that they are exposed, then for practically any point in the scene we can directly connect to it.

**Reply by coult099** [reply]
Thank you for the reply and resources!

**Comment by LowRanker** [comment]


**Reply by merlinND** [reply]
Hello @Jijiang1234,
Please open a new Discussion about your issue, making sure to fill the template to provide as much detail as possible about the problem your are facing.
